src/components/PostList.js

```diff
@@ -7,70 +7,117 @@ import {
 	Heading,
 	useTheme,
 	Spinner,
+	Button,
+	HStack,
+	useDisclosure,
+	Tooltip,
+	Flex,
 } from '@chakra-ui/react';
-import { getPosts } from '../api-client/ApiClient';
+import { ChevronRightIcon, ChevronLeftIcon } from '@chakra-ui/icons';
+import { getPosts, getAllPosts } from '../api-client/ApiClient';
 import { Link } from 'react-router-dom';
+import { isLoggedIn } from './Utils.js';
+import CustomButton from './Button';
+import NewConversationDrawer from './NewConversationDrawer';
 
-const isLoggedIn = () => {
-	return (
-		localStorage.getItem('logic_token') &&
-		localStorage.getItem('db_token') &&
-		localStorage.getItem('user_id')
-	);
-};
-
-const PostList = () => {
+const PostList = ({ usersRides }) => {
 	const theme = useTheme();
+	const { isOpen, onOpen, onClose } = useDisclosure();
 	const [posts, setPosts] = useState([]);
 	const [loading, setLoading] = useState(true);
 	const [error, setError] = useState('');
 	const [loggedIn] = useState(isLoggedIn());
+	const [currentPage, setCurrentPage] = useState(1);
+	const [selectedPost, setSelectedPost] = useState(null);
+	const postsPerPage = 3;
 
 	useEffect(() => {
-		if (loggedIn) {
+		if (loggedIn || !usersRides) {
 			const userId = localStorage.getItem('user_id');
 			const fetchPosts = async () => {
-				const response = await getPosts(userId);
-				if (response.error) {
-					setError(response.error);
-				} else {
-					setPosts(response);
+				setLoading(true);
+				let response = null;
+
+				try {
+					if (usersRides) {
```

Comment:
Not sure I understand this?

Reply:
We're using the same `posList` component for both user's own posts and the all posts which will display on home page. If the userRides argument passed into the component is true we will send `getPosts` request otherwise `getAllPosts` request.